spec/lib/karafka/instrumentation/logger_spec.rb

```diff
@@ -3,54 +3,94 @@
 RSpec.describe Karafka::Instrumentation::Logger do
   subject(:logger) { described_class.new }
 
+  let(:delegate_scope) { class_double(Karafka::Helpers::MultiDelegator) }
+
   # We use a singleton logger that could be already initialized in other specs, so
   # in order to check all the behaviours we need to "reset" it to the initial state
-  before { logger.instance_variable_set('@file', nil) }
+  before do
+    allow(Karafka::Helpers::MultiDelegator).to receive(:delegate)
+      .with(:write, :close)
+      .and_return(delegate_scope)
+    allow(delegate_scope).to receive(:to)
+  end
 
   specify { expect(described_class).to be < ::Logger }
 
   describe '#new' do
-    let(:target) { double }
-    let(:log_file) { Karafka::App.root.join('log', "#{Karafka.env}.log") }
-    # A Pathname, because this is what is returned by File.join
-    let(:log_dir) { File.dirname(log_file) }
-    let(:parent_dir) { File.dirname(log_dir) }
+    let(:karafka_root) { Pathname(Dir::Tmpname.create('karafka') { |_| }) }
```

Comment:
would you mind renaming it to something like `karafka_test_root`? `karafka_root` may be misleading as it would indicate the karafka app location.

Reply:
Done, CI is green.